### ISSUE TITLE
Missing owner reference in ingress created for devfile component

### ISSUE THREAD

**Issue by yangcao77** [openshift/odo#2808]
/kind bug

owner reference is missing in ingress spec; therefore when deleting the deployment, the ingress still remains. 

Details in : https://github.com/openshift/odo/pull/2762#issuecomment-608243209

**Comment by maysunfaisal** [comment]
The secrets would also need to be deleted along with the ingress:

```
secret/maysunopenliberty-tlssecret                                 kubernetes.io/tls                     2      17h   app.kubernetes.io/instance=maysunopenliberty,app.kubernetes.io/managed-by-version=v1.1.1,app.kubernetes.io/managed-by=odo,app.kubernetes.io/part-of=,app=
secret/maysunspring-tlssecret                                      kubernetes.io/tls                     2      17h   app.kubernetes.io/instance=maysunspring,app.kubernetes.io/managed-by-version=v1.1.1,app.kubernetes.io/managed-by=odo,app.kubernetes.io/part-of=,app=
```

**Comment by yangcao77** [comment]
@maysunfaisal The tls secrets need to be deleted during `odo delete`, not `odo url delete`. Since we are sharing the same tlssecret for all secured URLs within the same component.

**Comment by maysunfaisal** [comment]
Okay, let me rephrase -
shouldn't ingress and secret both be deleted during `odo delete`? Correct me if I am wrong, `odo url delete` currently deletes only the envinfo which is correct and matches `odo url create` since it creates envinfo..

**Comment by yangcao77** [comment]
yes. they should be both deleted during `odo delete`.
I guess what you mean is the owner reference should also be added to the tls-secrent? So that it can be deleted during `odo delete`.
But `odo url delete` followed with `odo push`, would not delete tls-secert, it only deletes ingress. So it's not be deleted "along with ingress", it's be deleted  "along with deploy"

**Comment by maysunfaisal** [comment]
@yangcao77 so; `odo url create` creates url envinfo and `odo url delete` will remove url envinfo  and ingress? because ingress is created during `odo push`?

**Comment by yangcao77** [comment]
@maysunfaisal `odo url delete` will remove url envinfo. 
And `odo push` would create/delete the ingress based on the difference between existing ingresses, and ingresses in envinfo.

**Comment by maysunfaisal** [comment]
@yangcao77 in that case what happens when i do `odo create`, `odo url create`, `odo push` and `odo delete`? 

The ingress would be left behind, since i am not doing `odo url delete` and `odo push` again to remove the ingress.. 🤔

**Comment by yangcao77** [comment]
@maysunfaisal `odo delete` would delete all ingresses and secrets belongs to that component. Since they are bonded to the same deployment. 
And the ingress and tls-secret is currently left behind because owner reference is missing. 
This is what the issue for.
